app/src/test/java/org/mozilla/fenix/TelemetryMiddlewareTest.kt

```diff
@@ -86,10 +91,13 @@ class TelemetryMiddlewareTest {
                 createTab(id = "2", url = "https://firefox.com"))
             )
         ).joinBlocking()
+        shadowOf(getMainLooper()).idle()
```

Comment:
nit: Hm why did you need these here? We should probably handle this with a test dispatcher instead?

Reply:
yes, needed to wait till idle.  Should I'll see if I can use a test dispatcher instead.